include/circt/Dialect/Handshake/HandshakePasses.td

```diff
@@ -62,6 +62,17 @@ def HandshakeRemoveBuffers : Pass<"handshake-remove-buffers", "handshake::FuncOp
   let constructor = "circt::handshake::createHandshakeRemoveBuffersPass()";
 }
 
+def HandshakeLowerExtmemToHW : Pass<"handshake-lower-extmem-to-hw", "mlir::ModuleOp"> {
```

Comment:
Could we change this to a function pass, instead of a module pass?

Reply:
... in theory yes, but I avoid making changes to the interface of a function-like op when a pass matches on that function (ideally, such a pass should also update all callers of the function-like op, which is only safe when the function matches on the scope where all definitions and callers are defined within). However, in this case, it _should_ be safe seeing as it is nevertheless illegal to instantiate other handshake functions with memref arguments.